### PR TITLE
refac: upstream latest plonky2

### PR DESCRIPTION
**Pull request by Insun35**
This PR includes changes to make it compatible with latest Plonky2.
- Add a condition for when the limb length are different that checks if all extra limbs are zero.
- Update plonky2-ecdsa dependency with patched version.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ ark-std = "0.4.0"
 num-bigint = "0.4.3"
 num-traits = "0.2"
 rand = "0.8.5"
-plonky2_ecdsa = { git = "https://github.com/Lagrange-Labs/plonky2-ecdsa", features = [
+plonky2_ecdsa = { git = "https://github.com/Lagrange-Labs/plonky2-ecdsa", rev = "5dd47af", features = [
```
Comment (silathdiir) [inline Cargo.toml]: May need to update after merging PR https://github.com/Lagrange-Labs/plonky2-ecdsa/pull/3.

**Review by silathdiir — APPROVED** [review]
LGTM!

File: src/fields/fq_target.rs [hunk 2]
```diff
@@ -107,27 +107,64 @@ impl<F: RichField + Extendable<D>, const D: usize> FqTarget<F, D> {
...
 
-        let terms = a_limbs
+        if a_limbs.len() == b_limbs.len() {
+            self.compare_same_length(builder, &a_limbs, &b_limbs)
+        } else {
+            self.compare_different_length(builder, &a_limbs, &b_limbs)
+        }
+    }
+
+    fn compare_same_length(
```
Comment (nikkolasg) [inline src/fields/fq_target.rs]: Sorry I fail to see why the two functions are different - if you give same length inputs to the `compare_different_lengths` then it should be returning the same no ? 
I think the only change required is to move `if lhs.len() < rhs.len()` to `if lhs.len() <= rhs.len()`.
In case it's equal `  for &limb in longer.iter().skip(shorter.len()) {` will never run so it'll give the same results no ?

TLDR do we really need this new function ?

**Review by nikkolasg — APPROVED** [review]
Approve for sake of time but left a comment, doesn't seem necessary to me this new function but maybe i'm missing something.
And https://github.com/Lagrange-Labs/plonky2-ecdsa/pull/3 have been merged so you can update deps 🙏

File: Cargo.toml (hunk 1 above)
Comment (Insun35) [reply Cargo.toml]: Updated in commit c37edcd. Thanks!

File: src/fields/fq_target.rs (hunk 2 above)
Comment (Insun35) [reply src/fields/fq_target.rs]: Makes sense, it could be simplified as you said. Fixed in commit 698cc66, thanks!
